不是重入循环，只是 catch 中特定的 `$exception->getCode()` ，简化日志，不然会是一大串报错的Stack trace

```
[ERROR] Swow\SocketException: Socket accept has been canceled in /www/wwwroot/vendor/swow/swow/lib/swow-
library/src/Psr7/Server/Server.php:46
Stack trace:
...
...
...
````

--- a/src/Http/Server.php
+++ b/src/Http/Server.php
@@ -87,7 +87,7 @@ class Server extends HttpServer implements ServerInterface
                     if (in_array($exception->getCode(), [Errno::EMFILE, Errno::ENFILE, Errno::ENOMEM], true)) {
                         $this->logger->warning('Socket resources have been exhausted.');
                         sleep(1);
-                    } elseif (in_array($exception->getCode(), [Errno::ECANCELED])) {
+                    } elseif (in_array($exception->getCode(), [Errno::ECANCELED], true)) {
                         $this->logger->notice('Socket accept has been canceled.');
                         break;
                     } else {

--- a/src/Http/Server.php
+++ b/src/Http/Server.php
@@ -84,9 +84,12 @@ class Server extends HttpServer implements ServerInterface
                         }
                     });
                 } catch (SocketException|CoroutineException $exception) {
-                    if (in_array($exception->getCode(), [Errno::EMFILE, Errno::ENFILE, Errno::ENOMEM, Errno::ECANCELED], true)) {
+                    if (in_array($exception->getCode(), [Errno::EMFILE, Errno::ENFILE, Errno::ENOMEM], true)) {
                         $this->logger->warning('Socket resources have been exhausted.');
                         sleep(1);
+                    } elseif (in_array($exception->getCode(), [Errno::ECANCELED])) {
+                        $this->logger->notice('Socket accept has been canceled');
+                        break;
                     } else {
                         $this->logger->error((string) $exception);
                         break;

--- a/src/Http/Server.php
+++ b/src/Http/Server.php
@@ -88,7 +88,7 @@ class Server extends HttpServer implements ServerInterface
                         $this->logger->warning('Socket resources have been exhausted.');
                         sleep(1);
                     } elseif (in_array($exception->getCode(), [Errno::ECANCELED])) {
-                        $this->logger->notice('Socket accept has been canceled');
+                        $this->logger->notice('Socket accept has been canceled.');
                         break;
                     } else {
                         $this->logger->error((string) $exception);

